oak-run/src/main/java/org/apache/jackrabbit/oak/indexversion/PurgeOldIndexVersion.java

```diff
@@ -38,43 +43,59 @@
+     *
+     * @throws IOException
+     * @throws CommitFailedException
+     */
+    public void execute(NodeStore nodeStore, boolean isReadWriteRepository, long purgeThresholdMillis, List<String> indexPaths) throws IOException, CommitFailedException {
         if (!isReadWriteRepository) {
-            LOG.info("Repository connected in read-only mode. Use '--read-write' for write operations");
+            LOG.info("Repository is opened in read-only mode");
+            return;
         }
-        try (NodeStoreFixture fixture = NodeStoreFixtureProvider.create(opts)) {
-            NodeStore nodeStore = fixture.getStore();
-            List<String> sanitisedIndexPaths = sanitiseUserIndexPaths(indexPaths);
-            Set<String> indexPathSet = filterIndexPaths(getRepositoryIndexPaths(nodeStore), sanitisedIndexPaths);
-            Map<String, Set<String>> segregateIndexes = segregateIndexes(indexPathSet);
-            for (Map.Entry<String, Set<String>> entry : segregateIndexes.entrySet()) {
-                String baseIndexPath = entry.getKey();
-                String parentPath = PathUtils.getParentPath(entry.getKey());
-                List<IndexName> indexNameObjectList = getIndexNameObjectList(entry.getValue());
-                NodeState indexDefParentNode = NodeStateUtils.getNode(nodeStore.getRoot(),
-                        parentPath);
-                List<IndexVersionOperation> toDeleteIndexNameObjectList =
-                        IndexVersionOperation.generateIndexVersionOperationList(indexDefParentNode, indexNameObjectList, purgeThresholdMillis);
-                if (isReadWriteRepository && !toDeleteIndexNameObjectList.isEmpty()) {
-                    purgeOldIndexVersion(nodeStore, toDeleteIndexNameObjectList);
-                } else {
-                    LOG.info("Repository is opened in read-only mode: IndexOperations" +
-                            " for index at path {} are : {}", baseIndexPath, toDeleteIndexNameObjectList);
-                }
+        List<IndexVersionOperation> purgeIndexList = getPurgeIndexes(nodeStore, purgeThresholdMillis, indexPaths);
+        if (!purgeIndexList.isEmpty()) {
+            LOG.info("Found indexes for purging: '{}'", purgeIndexList);
+            long purgeStart = System.currentTimeMillis();
+            purgeOldIndexVersion(nodeStore, purgeIndexList);
+            LOG.info("Index purging done, took '{}' ms", (System.currentTimeMillis() - purgeStart));
+        } else {
+            LOG.info("No indexes are found to be purged");
+        }
+    }
+
+    public List<IndexVersionOperation> getPurgeIndexes(NodeStore nodeStore, long purgeThresholdMillis, List<String> indexPaths) throws IOException, CommitFailedException {
+        List<IndexVersionOperation> purgeIndexList = new ArrayList<>();
+        LOG.info("Start execute purge index over node type '{}' over index paths '{}'", nodeStore.getClass().getSimpleName(), indexPaths);
```

Comment:
I would log "Getting indexes to purge ..." as in this method we don't purge.

Reply:
@thomasmueller thanks, makes sense, will change it.